TWLight/resources/tests.py

```diff
@@ -501,6 +501,35 @@ def test_user_instructions(self):
             # We're making sure that it fails because of a lack of user instructions, specifically.
             self.assertEqual([error_msg], e.messages)
 
+    def test_for_target_url_not_empty_when_authorization_method_is_proxy_or_bundle(
+        self,
+    ):
+        """
+        When partner’s authorization method is PROXY or BUNDLE
+        then leaving target_url empty should raise a ValidationError
+        """
+        message = "Proxy and Bundle partners require a target URL."
+
+        # test for a partner with proxy authorization method
+        partner1 = PartnerFactory(authorization_method=Partner.PROXY)
+        partner1.requested_access_duration = (
+            True  # We don't want the ValidationError from requested_access_duration
+        )
+
+        self.assertRaises(ValidationError, partner1.clean)
+        try:
+            partner1.save()
+        except ValidationError as e:
+            self.assertEqual([message], e.messages)
+
+        # test for a partner with proxy authorization method
+        partner2 = PartnerFactory(authorization_method=Partner.BUNDLE)
+        self.assertRaises(ValidationError, partner2.clean)
+        try:
+            partner2.save()
+        except ValidationError as e:
+            self.assertEqual([message], e.messages)
+
```

Comment:
I think it would also be helpful (in light of a recent bug we found - https://phabricator.wikimedia.org/T265331) to have another test which checks that for a non-PROXY or BUNDLE partner, no error message is raised.